app/components/NavModal/index.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
+
+const NavModalDialog = styled.div.attrs({
+  className: 'modal-dialog',
+})`
+  &&& {
+    font-size: 2rem;
+    font-weight: 600;
+  }
+`;
+
+const NavModalContent = styled.div.attrs({
+  className: 'modal-content',
+})`
+  &&& {
+    background: transparent;
+    border: transparent;
+
+    & > div {
+      border: inherit;
+    }
+  }
+`;
+
+const ModalCloseX = styled.span`
+  color: #ddd;
+  font-size: 3.25rem;
+  text-shadow: none;
+`;
+
+const ModalLink = styled(Link)`
+  &&& {
+    color: #b8daff;
+  }
+`;
+
+function NavModal() {
+  // Callback for click event: Modal needs help closing properly after clicking on link (otherwise overlay will stay and make site unusable).
+  function hideNavModal() {
+    document.getElementById('nav-modal-close-btn').click();
```

Comment:
Please use `data-dismiss="modal"` on the close button instead of custom jquery here. Always want to rely on bootstrap instead of custom JS

Reply:
https://getbootstrap.com/docs/4.0/components/modal/

Reply:
Definitely tried to do that. But this code makes sure the modal closes properly when changing pages (clicking a link). Adding `data-dismiss="modal"` to the modal body or the elements within doesn't work. This still relies on Bootstrap's behavior by just replicating a click on the close button.